### PR TITLE
Feature/wb 14/list of templates with screenshot image and description

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

File: src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        <styled.Description
+          text={template.description}
+          size="m"
+          isMultiline={selected}
+          align="left"
+          className={cn(selected && 'selected')}
+        />
+      </styled.InfoContainer>
+      <styled.Image
+        src={`${appVariables.RENDER_SERVICE_URL}/get/${template.image}`}
```
Comment (aminbksi) [inline src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx:48]: maybe move this to store

File: src/scenes/worldBuilder/pages/WorldBuilderNamePage/WorldBuilderNamePage.tsx [hunk 2]
```diff
@@ -30,21 +33,23 @@ const WorldBuilderNamePage: FC = () => {
...
     const {valid: subdomainIsValid, error: subdomainError} =
-      await worldNameStore.validateSubdomainName(data.subdomainName);
+      await worldBuilderNameStore.validateSubdomainName(data.subdomainName);
 
     if (nameIsValid && subdomainIsValid) {
-      // TODO: redirect to next step
-      console.info('Redirecting to template...');
+      worldBuilderNameStore.submit(data.name, data.subdomainName);
+      history.push(ROUTES.worldBuilder.template);
       return;
```
Comment (aminbksi) [inline src/scenes/worldBuilder/pages/WorldBuilderNamePage/WorldBuilderNamePage.tsx:53]: does it need a return?

File: src/api/repositories/worldBuilderRepository/worldBuilderRepository.api.types.ts [hunk 3]
```diff
@@ -1,3 +1,13 @@
+export interface TemplateInterface {
+  id: string;
+  name: string;
+  description: string;
+  image: string;
+  content: object;
```
Comment (PAKovalev) [inline src/api/repositories/worldBuilderRepository/worldBuilderRepository.api.types.ts:6]: Is there special type?

File: src/scenes/worldBuilder/components/WorldBuilderHeader/WorldBuilderHeader.tsx [hunk 4]
```diff
@@ -1,17 +1,27 @@
...
+}
+
+const WorldBuilderHeader: FC<PropsInterface> = ({size = 'normal'}) => {
   const {t} = useTranslation();
 
   return (
-    <styled.LogoContainer>
-      <styled.Logo src={logo} />
-      <styled.Title label={t('titles.worldBuilder')} transform="uppercase" type="h1" />
+    <styled.LogoContainer className={cn(size)}>
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/components/WorldBuilderHeader/WorldBuilderHeader.tsx:17]: Remove `cn`

File: src/scenes/worldBuilder/components/WorldBuilderHeader/WorldBuilderHeader.tsx [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
+      <styled.Logo src={logo} className={cn(size)} />
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/components/WorldBuilderHeader/WorldBuilderHeader.tsx:18]: `cn`

File: src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.styled.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+import {Text} from 'ui-kit';
+
+export const Container = styled.div`
+  border-radius: 10px;
+  overflow: hidden;
+  position: relative;
+  border: 1px solid transparent;
+
+  :not(&.selected) {
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.styled.ts:11]: Does it work? `&:not`

File: src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import * as styled from './WorldBuilderTemplateItem.styled';
+
+interface PropsInterface {
+  template: WorldBuilderTemplateModelInterface;
+  selected?: boolean;
+  onClick?: (template: WorldBuilderTemplateModelInterface) => void;
+}
+
+const WorldBuilderTemplateItem: FC<PropsInterface> = ({template, onClick, selected = false}) => {
+  const ref = useRef(null);
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx:17]: type of `ref`

File: src/scenes/worldBuilder/pages/WorldBuilderNamePage/WorldBuilderNamePage.tsx (hunk 2 above)
Comment (Daboo1998) [reply src/scenes/worldBuilder/pages/WorldBuilderNamePage/WorldBuilderNamePage.tsx:53]: Yes and no, but why do the next lines if everything is valid (so no errors)? Also there might be other stuff done if its not valid, and you may want that early return then.

File: src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx (hunk 1 above)
Comment (PAKovalev) [inline src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx]: Move to model

File: src/scenes/worldBuilder/pages/WorldBuilderGeneratePage/WorldBuilderGeneratePage.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    <Page backgroundSrc={background} showSimpleProfileMenu>
+      <styled.Container>
+        <styled.Spacer />
+        <styled.TemplateContainer>
+          <WorldBuilderTemplateItem
+            template={worldBuilderTemplatesStore.selectedTemplate}
+            selected
+          />
+        </styled.TemplateContainer>
+        <WorldBuilderFooter currentStep={2} buttonLabel={t('actions.generateWorld')} />
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/pages/WorldBuilderGeneratePage/WorldBuilderGeneratePage.tsx:31]: Make constant for `2`

File: src/scenes/worldBuilder/pages/WorldBuilderTemplatePage/WorldBuilderTemplatePage.tsx [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+import React, {FC, useEffect} from 'react';
+import {useHistory} from 'react-router-dom';
+
+import {ROUTES} from 'core/constants';
+import {
+  WorldBuilderFooter,
+  WorldBuilderHeader,
+  WorldBuilderTemplateItem
+} from 'scenes/worldBuilder/components';
+import {WorldBuilderTemplateModelInterface} from 'scenes/worldBuilder/stores/WorldBuilderTemplatesStore/models';
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/pages/WorldBuilderTemplatePage/WorldBuilderTemplatePage.tsx:11]: Do you really need to export this interface?

File: src/scenes/worldBuilder/stores/WorldBuilderNameStore/WorldBuilderNameStore.ts [hunk 10]
```diff
@@ -31,7 +33,12 @@ const WorldNameStore = types
       );
 
       return response;
-    })
+    }),
+    submit(name: string, subdomain: string) {
+      self.resetModel();
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/stores/WorldBuilderNameStore/WorldBuilderNameStore.ts:38]: Usually `resetModel` is part of `useEffect`. It is more clear

File: src/scenes/worldBuilder/stores/WorldBuilderTemplatesStore/WorldBuilderTemplatesStore.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import {cast, flow, types} from 'mobx-state-tree';
+
+import {api} from 'api';
+import {TemplatesResponse} from 'api/repositories/worldBuilderRepository/worldBuilderRepository.api.types';
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/stores/WorldBuilderTemplatesStore/WorldBuilderTemplatesStore.ts:4]: `import {TemplatesResponse} from 'api`;`

File: src/scenes/worldBuilder/stores/WorldBuilderTemplatesStore/models/WorldBuilderTemplateModel/WorldBuilderTemplateModel.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
+import {Instance, types} from 'mobx-state-tree';
+
+const WorldBuilderTemplateModel = types
```
Comment (PAKovalev) [inline src/scenes/worldBuilder/stores/WorldBuilderTemplatesStore/models/WorldBuilderTemplateModel/WorldBuilderTemplateModel.ts:5]: No need to use `Model` anymore.

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: src/api/repositories/worldBuilderRepository/worldBuilderRepository.api.types.ts (hunk 3 above)
Comment (Daboo1998) [reply src/api/repositories/worldBuilderRepository/worldBuilderRepository.api.types.ts:6]: No, but let me check (there is no really documentation on that)

File: src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx (hunk 7 above)
Comment (Daboo1998) [reply src/scenes/worldBuilder/components/WorldBuilderTemplateItem/WorldBuilderTemplateItem.tsx:17]: Actually, have to remove it, its not needed

File: src/scenes/worldBuilder/pages/WorldBuilderTemplatePage/WorldBuilderTemplatePage.tsx (hunk 9 above)
Comment (Daboo1998) [reply src/scenes/worldBuilder/pages/WorldBuilderTemplatePage/WorldBuilderTemplatePage.tsx:11]: Yes cause I need to access the whole template in a different page

File: src/api/repositories/worldBuilderRepository/worldBuilderRepository.api.types.ts (hunk 3 above)
Comment (Daboo1998) [reply src/api/repositories/worldBuilderRepository/worldBuilderRepository.api.types.ts:6]: Removed it

File: src/scenes/worldBuilder/pages/WorldBuilderNamePage/WorldBuilderNamePage.tsx (hunk 2 above)
Comment (PAKovalev) [reply src/scenes/worldBuilder/pages/WorldBuilderNamePage/WorldBuilderNamePage.tsx:56]: Actually, you use this interface only in this line:
`(template: WorldBuilderTemplateModelInterface)`.

But it is not necessary.

**Review by PAKovalev — APPROVED** [review]

**Pull request merged**